File: app/src/main/java/com/jeksonshar/jweather/root/MainActivity.java
package com.jeksonshar.jweather.root;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

import com.jeksonshar.jweather.R;
import com.jeksonshar.jweather.ui.WeatherFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.fragment_container);

        if (fragment == null) {
            fragment = new WeatherFragment();

            fm.beginTransaction().add(R.id.fragment_container, fragment).commit();
        }
    }
}

File: app/src/main/java/com/jeksonshar/jweather/root/MainActivity.kt
package com.jeksonshar.jweather.root

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.jeksonshar.jweather.R
import com.jeksonshar.jweather.ui.WeatherFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(R.id.fragment_container)

        if (fragment == null) {
            fragment = WeatherFragment()
            fm.beginTransaction().add(R.id.fragment_container, fragment).commit()
        }
    }
}